### PR TITLE
[Rest Server] Add get job api v2

### PR DESCRIPTION
**Pull request by abuccts**
Add get job api v2.

### CODE REVIEW

File: src/rest-server/src/controllers/v2/job.js [hunk 1]
```diff
@@ -17,29 +17,29 @@
...
   try {
-    const getJobError = await util.promisify(v1Job.prototype.getJob)(jobName, userName);
-    if (getJobError.code !== 'NoJobError') {
+    const data = await job.get(frameworkName);
+    if (data != null) {
       throw createError('Conflict', 'ConflictJobError', `Job ${frameworkName} already exists.`);
     }
   } catch (error) {
-    throw createError('Conflict', 'ConflictJobError', `Job ${frameworkName} already exists.`);
+    if (error.code === 'UnknownError') {
```
Comment (Binyang2014) [inline src/rest-server/src/controllers/v2/job.js]: Only throw error when error code is UnknownError? It's an expect behavior?

Comment (abuccts) [reply src/rest-server/src/controllers/v2/job.js]: yes

Comment (abuccts) [reply src/rest-server/src/controllers/v2/job.js]: updated

File: src/rest-server/src/models/v2/job.js [hunk 2]
```diff
@@ -271,10 +271,13 @@ async function get(frameworkName) {
     url: launcherConfig.frameworkPath(frameworkName),
     headers: launcherConfig.webserviceRequestHeaders(userName),
   });
-  if (response.status !== status('OK')) {
+  if (response.status === status('OK')) {
+    return response.data;
+  } else if (response.status === status('Not Found')) {
```
Comment (Binyang2014) [inline src/rest-server/src/models/v2/job.js]: remove else
```js
if (status === statue('OK')) {
  return ...;
}

if (...) {
  throw ...
}
```

File: src/rest-server/src/controllers/v2/job.js [hunk 3]
```diff
@@ -35,7 +35,7 @@ const update = asyncHandler(async (req, res) => {
       throw createError('Conflict', 'ConflictJobError', `Job ${frameworkName} already exists.`);
     }
   } catch (error) {
-    if (error.code === 'UnknownError') {
+    if (error.code !== 'NoJobError') {
```
Comment (Binyang2014) [inline src/rest-server/src/controllers/v2/job.js]: Can log this?
```js
if (error.code === 'NoJobError') {
  log.Debug('...');
  return;
}

throw error;
```

File: src/rest-server/src/models/v2/job.js (hunk 2 above)
Comment (abuccts) [reply src/rest-server/src/models/v2/job.js]: updated

File: src/rest-server/src/controllers/v2/job.js (hunk 3 above)
Comment (abuccts) [reply src/rest-server/src/controllers/v2/job.js]: it's not an error if job not found here, no need to log.

**Pull request merged**